### PR TITLE
opencascade: Update to 7.6.0

### PR DESCRIPTION
**Pull request by taoyouh**
Specify library name and version:  **opencascade/7.6.0**

Updates:
- Update opencascade recipe for 7.6.0 version
- Add ffmepg support for the package
- Add new with_tk option to allow building without TK (new in 7.6.0)
- Add new with_opengl option to allow building without OpenGL (new in 7.6.0)
- Add optional dependency draco for 7.6.0.

Questions:
- Should the default value of with_tk be True or False? The CMakeLists.txt define it as ON by default, but it is currently set to False to match the other settings.

Further works:
- The build [doc](https://dev.opencascade.org/doc/occt-7.6.0/overview/html/build_upgrade__building_occt.html) says FreeType is now an optional dependency, but a [bug](https://tracker.dev.opencascade.org/view.php?id=31087) makes it actually mandatory. We will need to wait until 7.6.1 to build without FreeType.
- Some (new and existing) long lines and "##" prefixed comments violates PEP8 rules.
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by taoyouh** [comment]
> Compiler is out of heap space

How do we usually fix this kind of problems?

**Comment by taoyouh** [comment]
My local build worked well on my laptop with 8GB RAM. But I'm using MSVC 19.29.30138.0 which is newer than that in the logs.

**Review by SSE4 — DISMISSED** [review]

**Review by SpaceIm — DISMISSED** [review]

**Comment by SpaceIm** [comment]
> Should the default value of with_tk be True or False? The CMakeLists.txt define it as ON by default, but it is currently set to False to match the other settings.

Missed that, it should be True yes, CCI recipes try to honor upstream default values.

> Some (new and existing) long lines and "##" prefixed comments violates PEP8 rules.

Don' worry for that 😉

File: recipes/opencascade/all/conanfile.py [hunk 1]
```diff
@@ -141,21 +169,34 @@ def _patch_sources(self):
...
+            if tools.Version(self.version) >= "7.6.0":
+                tools.replace_in_file(
+                    occt_csf_cmake,
+                    "set (CSF_fontconfig \"fontconfig\")",
+                    "set (CSF_fontconfig \"{}\")".format(" ".join(self.deps_cpp_info["fontconfig"].libs)))
+            else:
+                tools.replace_in_file(
+                    occt_csf_cmake,
+                    "set (CSF_fontconfig  \"fontconfig\")",
+                    "set (CSF_fontconfig  \"{}\")".format(" ".join(self.deps_cpp_info["fontconfig"].libs)))
```
Comment (prince-chrismc) [inline recipes/opencascade/all/conanfile.py:199]: These look exactly the same 🤔 can someone sanity check me?

I might need more coffee

**Review by prince-chrismc — COMMENTED** [review]
One tiny question for me

File: recipes/opencascade/all/conanfile.py (hunk 1 above)
Comment (taoyouh) [reply recipes/opencascade/all/conanfile.py:199]: Difference: number of spaces after `CSF_fontconfig`

Upstream changed the indentation and they try to align each part of the code.

Comment (prince-chrismc) [reply recipes/opencascade/all/conanfile.py:199]: Ohh! 🤢

Thanks for pointing that out 🤗

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**